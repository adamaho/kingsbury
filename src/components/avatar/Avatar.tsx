import * as React from 'react';
import styled from 'styled-components';

export interface AvatarProps {
  /** classname for the avatar */
  className?: string;

  /** src for the avatar */
  src?: React.ReactNode;
}

const Container = styled.div`
  height: 40px;
  width: 40px;

  display: flex;
  align-items: center;
  justify-content: center;

  border-radius: 50%;
  background-color: lightgrey;
`;

export const Avatar: React.FunctionComponent<AvatarProps> = (props) => {
  const {
    src,
    className
  } = props;

  return (
    <Container className={className}>
      {src}
    </Container>
  );
};

Avatar.defaultProps = {
  className: ''
};

export default Avatar;
