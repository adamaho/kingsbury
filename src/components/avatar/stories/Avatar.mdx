import { Story, Preview, Props } from '@storybook/addon-docs/blocks';
import { Title, SectionTitle } from '../../../../.storybook/components';
import { Avatar } from '../Avatar';

<Title
  title="Avatar"
  subtitle="Avatars allow users represent a user."
/>

<SectionTitle>Source</SectionTitle>

Using the source prop, you can pass a `ReactNode` to display in the middle of the avatar.

<Preview>
  <Story id="components-avatar--source"/>
</Preview>

<SectionTitle>Props</SectionTitle>
<Props of={Avatar} />
