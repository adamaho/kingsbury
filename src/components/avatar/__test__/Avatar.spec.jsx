import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from '..';

describe('Avatar', () => {
  test('it renders', () => {
    const tree = renderer.create(<Avatar src={<div>asdf</div>} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
