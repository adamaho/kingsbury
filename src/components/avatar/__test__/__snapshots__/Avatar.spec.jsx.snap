// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar it renders 1`] = `
.c0 {
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  <div>
    asdf
  </div>
</div>
`;
